- Add missing dependency for aom_static used by ENABLE_EXAMPLES
- Drop unused static libhwy dependency when using shared libjxl

--- CMakeLists.txt.orig	2023-08-14 21:22:47 UTC
+++ CMakeLists.txt
@@ -568,17 +568,14 @@ if(CONFIG_AV1_ENCODER)
       target_include_directories(aom_dsp_encoder PRIVATE ${LIBJXL_INCLUDE_DIRS})
     else()
       pkg_check_modules(LIBJXL REQUIRED libjxl)
+      if(BUILD_SHARED_LIBS)
+        target_link_libraries(aom_static
+                              PRIVATE ${LIBJXL_LDFLAGS} ${LIBJXL_LIBRARIES})
+      endif()
       target_link_libraries(aom PRIVATE ${LIBJXL_LDFLAGS} ${LIBJXL_LIBRARIES})
       target_include_directories(aom_dsp_encoder PRIVATE ${LIBJXL_INCLUDE_DIRS})
       if(LIBJXL_CFLAGS)
         append_compiler_flag("${LIBJXL_CFLAGS}")
-      endif()
-      pkg_check_modules(LIBHWY REQUIRED libhwy)
-      target_link_libraries(aom PRIVATE ${LIBHWY_LDFLAGS} ${LIBHWY_LIBRARIES})
-      target_include_directories(aom_dsp_encoder
-                                 PRIVATE ${LIBLIBHWY_INCLUDE_DIRS})
-      if(LIBHWY_CFLAGS)
-        append_compiler_flag("${LIBHWY_CFLAGS}")
       endif()
     endif()
 
